compiler+runtime/src/cpp/jank/runtime/obj/persistent_string.cpp

```diff
@@ -99,6 +99,16 @@ namespace jank::runtime::obj
     throw std::runtime_error{ fmt::format("get_entry not supported on string") };
   }
 
+  object_ptr persistent_string::nth(object_ptr const index) const
+  {
+    return get(index);
```

Comment:
This needs to throw if out-of-bounds.

Reply:
Note for Bill: `get` doesn't throw, for strings. It just returns `nil`. But `nth` does.

```
user=> (nth "foo" 5)
Execution error (StringIndexOutOfBoundsException) at jdk.internal.util.Preconditions$1/apply (Preconditions.java:55).
Index 5 out of bounds for length 3
user=> (get "foo" 5)
nil
```

We can just check the result of `get` and throw if it's `nil`. Let's format a message which helpfully provides the index and also the string's size. We do this already in `substring`.